### PR TITLE
[new release] ipaddr-cstruct, macaddr, macaddr-sexp, ipaddr, ipaddr-sexp and macaddr-cstruct (4.0.0)

### PR DESCRIPTION
**Pull request by avsm**
A library for manipulation of IP address representations using Cstructs

- Project page: <a href="https://github.com/mirage/ocaml-ipaddr">https://github.com/mirage/ocaml-ipaddr</a>
- Documentation: <a href="https://mirage.github.io/ocaml-ipaddr/">https://mirage.github.io/ocaml-ipaddr/</a>

##### CHANGES:

* Rename the `to/from_bytes` functions to refer to `octets`
  instead.  This distinguishes the meaning of human-readable
  addresses (`string`s in this library) and byte-packed
  representations(`octet`s in this library) from the OCaml
  `bytes` type that represents mutable strings.

  Porting code should just be a matter of renaming functions
  such as:
   - `Ipaddr.of_bytes` becomes `Ipaddr.of_octets`
   - `Macaddr.to_bytes` becomes `Macaddr.to_octets`

* Introduce new `write_octets` functions that can write
  octet representations of IPv4/v6 into an existing bytestring.

* Use the `domain-name` library to produce domain names
  from IP addresses.

* Remove the `ipaddr.sexp` and `macaddr.sexp` ocamlfind
  subpackages and instead have `ipaddr-sexp` and `macaddr-sexp`
  to match the opam package names.

* Add new `Ipaddr_cstruct` and `Macaddr_cstruct` libraries
  for conversion to/from cstructs (mirage/ocaml-ipaddr#36 @nojb @avsm)

### CODE REVIEW

**Comment by avsm** [comment]
This will need quite a few  upper bounds

File: packages/ipaddr-cstruct/ipaddr-cstruct.4.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+authors: ["David Sheets" "Anil Madhavapeddy" "Hugo Heuzard"]
+synopsis: "A library for manipulation of IP address representations using Cstructs"
+license: "ISC"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-ipaddr"
+doc: "https://mirage.github.io/ocaml-ipaddr/"
+bug-reports: "https://github.com/mirage/ocaml-ipaddr/issues"
+depends: [
+  "ocaml" {>= "4.04.0"}
+  "dune" {build}
```
Comment (kit-ty-kate) [inline packages/ipaddr-cstruct/ipaddr-cstruct.4.0.0/opam]:
```suggestion
  "dune"
```

File: packages/ipaddr-sexp/ipaddr-sexp.4.0.0/opam [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+"""
+
+license: "ISC"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-ipaddr"
+doc: "https://mirage.github.io/ocaml-ipaddr/"
+bug-reports: "https://github.com/mirage/ocaml-ipaddr/issues"
+depends: [
+  "ocaml" {>= "4.04.0"}
+  "dune" {build}
```
Comment (kit-ty-kate) [inline packages/ipaddr-sexp/ipaddr-sexp.4.0.0/opam]:
```suggestion
  "dune" {>= "1.9"}
```

File: packages/ipaddr/ipaddr.4.0.0/opam [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+"""
+
+license: "ISC"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-ipaddr"
+doc: "https://mirage.github.io/ocaml-ipaddr/"
+bug-reports: "https://github.com/mirage/ocaml-ipaddr/issues"
+depends: [
+  "ocaml" {>= "4.04.0"}
+  "dune" {build}
```
Comment (kit-ty-kate) [inline packages/ipaddr/ipaddr.4.0.0/opam]:
```suggestion
  "dune" {>= "1.9"}
```

File: packages/macaddr-cstruct/macaddr-cstruct.4.0.0/opam [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+authors: ["David Sheets" "Anil Madhavapeddy" "Hugo Heuzard"]
+synopsis: "A library for manipulation of MAC address representations using Cstructs"
+license: "ISC"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-ipaddr"
+doc: "https://mirage.github.io/ocaml-ipaddr/"
+bug-reports: "https://github.com/mirage/ocaml-ipaddr/issues"
+depends: [
+  "ocaml" {>= "4.04.0"}
+  "dune" {build}
```
Comment (kit-ty-kate) [inline packages/macaddr-cstruct/macaddr-cstruct.4.0.0/opam]:
```suggestion
  "dune" {>= "1.9"}
```

File: packages/macaddr-sexp/macaddr-sexp.4.0.0/opam [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+authors: ["David Sheets" "Anil Madhavapeddy" "Hugo Heuzard"]
+synopsis: "A library for manipulation of MAC address representations using sexp"
+license: "ISC"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-ipaddr"
+doc: "https://mirage.github.io/ocaml-ipaddr/"
+bug-reports: "https://github.com/mirage/ocaml-ipaddr/issues"
+depends: [
+  "ocaml" {>= "4.04.0"}
+  "dune" {build}
```
Comment (kit-ty-kate) [inline packages/macaddr-sexp/macaddr-sexp.4.0.0/opam]:
```suggestion
  "dune" {>= "1.9"}
```

File: packages/macaddr/macaddr.4.0.0/opam [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+authors: ["David Sheets" "Anil Madhavapeddy" "Hugo Heuzard"]
+synopsis: "A library for manipulation of MAC address representations"
+license: "ISC"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-ipaddr"
+doc: "https://mirage.github.io/ocaml-ipaddr/"
+bug-reports: "https://github.com/mirage/ocaml-ipaddr/issues"
+depends: [
+  "ocaml" {>= "4.04.0"}
+  "dune" {build}
```
Comment (kit-ty-kate) [inline packages/macaddr/macaddr.4.0.0/opam]:
```suggestion
  "dune" {>= "1.9"}
```

**Pull request merged**